app/src/main/java/de/danoeh/antennapod/fragment/CompletedDownloadsFragment.java

```diff
@@ -180,10 +184,40 @@ public boolean onMenuItemClick(MenuItem item) {
         } else if (item.getItemId() == R.id.action_search) {
             ((MainActivity) getActivity()).loadChildFragment(SearchFragment.newInstance());
             return true;
+        }  else if (item.getItemId() == R.id.downloads_sort_date_asc) {
+            setSortOrder(SortOrder.DATE_OLD_NEW);
+            return true;
+        } else if (item.getItemId() == R.id.downloads_sort_date_desc) {
+            setSortOrder(SortOrder.DATE_NEW_OLD);
+            return true;
+        }  else if (item.getItemId() == R.id.downloads_sort_duration_asc) {
+            setSortOrder(SortOrder.DURATION_SHORT_LONG);
+            return true;
+        } else if (item.getItemId() == R.id.downloads_sort_duration_desc) {
+            setSortOrder(SortOrder.DURATION_LONG_SHORT);
+            return true;
+        } else if (item.getItemId() == R.id.downloads_sort_episode_title_asc) {
+            setSortOrder(SortOrder.EPISODE_TITLE_A_Z);
+            return true;
+        } else if (item.getItemId() == R.id.downloads_sort_episode_title_desc) {
+            setSortOrder(SortOrder.EPISODE_TITLE_Z_A);
+            return true;
+        } else if (item.getItemId() == R.id.downloads_sort_feed_title_asc) {
+            setSortOrder(SortOrder.FEED_TITLE_A_Z);
+            return true;
+        } else if (item.getItemId() == R.id.downloads_sort_feed_title_desc) {
+            setSortOrder(SortOrder.FEED_TITLE_Z_A);
+            return true;
```

Comment:
We might want to add sorting to even more screens. Copying this list to all of them feels redundant. How about creating a class that "converts" a menu item ID to a SortOrder enum? Then that can be used on all screens.